Neos.ContentRepository.Core/Classes/Service/ContentStreamPruner.php

```diff
@@ -73,28 +75,28 @@ public function prune(bool $removeTemporary = false): iterable
      *
      *   - Otherwise, we cannot replay the other content streams correctly (if the base content streams are missing).
      *
-     * @return ContentStreams the removed content streams
+     * @return list<StreamName> the removed content streams
      */
-    public function pruneRemovedFromEventStream(): ContentStreams
+    public function pruneRemovedFromEventStream(): array
     {
-        $removedContentStreams = $this->findUnusedAndRemovedContentStreams();
-        foreach ($removedContentStreams as $removedContentStream) {
-            $streamName = ContentStreamEventStreamName::fromContentStreamId($removedContentStream->id)
-                ->getEventStreamName();
-            $this->eventStore->deleteStream($streamName);
+        $removedContentStreams = $this->findUnusedAndRemovedContentStreamIds();
+        foreach ($removedContentStreams as $removedContentStreamName) {
+            $this->eventStore->deleteStream($removedContentStreamName);
         }
         return $removedContentStreams;
     }
 
     public function pruneAll(): void
     {
-        foreach ($this->contentRepository->findContentStreams() as $contentStream) {
-            $streamName = ContentStreamEventStreamName::fromContentStreamId($contentStream->id)->getEventStreamName();
+        foreach ($this->findAllContentStreamEventNames() as $streamName) {
             $this->eventStore->deleteStream($streamName);
         }
     }
 
-    private function findUnusedAndRemovedContentStreams(): ContentStreams
+    /**
+     * @return list<StreamName>
+     */
+    private function findUnusedAndRemovedContentStreamIds(): array
```

Comment:
todo maybe not use the projection here at all to ensure the prune can always be run?